common/aws/s3/client.go

```diff
@@ -105,15 +106,32 @@ func NewClient(ctx context.Context, cfg commonaws.ClientConfig, logger logging.L
 	return ref, err
 }
 
+func PeekObjectSize(ctx context.Context, s3Client *s3.Client, bucket, key string) (int64, error) {
```

Comment:
We already have this method in this file `HeadObject`. Could we have used that one?